src/utils.js

```diff
@@ -80,89 +75,36 @@ function returnNeighbouringLadCode(ladCode, searchLowerLadCode) {
-  const total = data.get(totalCatCode);
-  populateCensusTable.categories.forEach((category) => {
-    if (data.has(category.code)) {
-      category["value"] = data.get(category.code);
-      category["percentage"] = (Math.round((category.value / total) * 100 * 10) / 10).toFixed(1);
-      category["value"] = category["value"].toLocaleString();
-    }
-  });
-  return populateCensusTable;
-}
-
-export function filterSelectedTable(metadata, category) {
-  let selectedTable;
-  metadata.forEach((topic) => {
-    topic.tables.forEach((table) => {
-      if (table.code == category.table) {
-        selectedTable = table;
-      }
-    });
-  });
-  return selectedTable;
-}
-
-export function populateSelectedCatData(geoCode, totalCatCode, tableSlug, categorySlug) {
-  if (get(dataByGeography).has(geoCode)) {
-    let category = getCategoryBySlug(tableSlug, categorySlug);
-    if (get(dataByGeography).get(geoCode).has(totalCatCode) && get(dataByGeography).get(geoCode).has(category.code)) {
-      return {
-        total: get(dataByGeography).get(geoCode).get(totalCatCode).toLocaleString(),
-        val: get(dataByGeography).get(geoCode).get(category.code).toLocaleString(),
-        perc: (
-          Math.round(
-            (get(dataByGeography).get(geoCode).get(category.code) /
-              get(dataByGeography).get(geoCode).get(totalCatCode)) *
-              100 *
-              10,
-          ) / 10
-        ).toFixed(1),
-        unit: filterSelectedTable(metadata, category).units,
-        geoCode: geoCode,
-      };
-    }
+export function populateSelectedCatData(geoCode, category) {
+  if (get(dataByGeography).has(geoCode) && get(dataByGeography).get(geoCode).has(category.code)) {
+    return {
+      total: get(dataByGeography).get(geoCode).get(category.code)["total"].toLocaleString(),
+      val: get(dataByGeography).get(geoCode).get(category.code)["value"].toLocaleString(),
+      perc: get(dataByGeography).get(geoCode).get(category.code)["perc"],
+      unit: tables[category.table].unit,
+      geoCode: geoCode,
+    };
   }
 }
 
-export function calculateComparisonDiff(geoCode, comparatorGeoCode, totalCatCode, category) {
-  let comparatorTotal;
-  let comparatorVal;
-  if (comparatorGeoCode == config.eAndWGeoCode) {
-    comparatorTotal = get(englandAndWalesData).get(config.eAndWGeoCode).get(totalCatCode);
-    comparatorVal = get(englandAndWalesData).get(config.eAndWGeoCode).get(category.code);
-  } else {
-    comparatorTotal = get(dataByGeography).get(comparatorGeoCode).get(totalCatCode);
-    comparatorVal = get(dataByGeography).get(comparatorGeoCode).get(category.code);
+export function calculateComparisonDiff(geoCode, comparatorGeoCode, catCode) {
+  if (get(dataByGeography).has(geoCode) && get(dataByGeography).get(geoCode).has(catCode)) {
+    let comparatorCategory;
+    if (comparatorGeoCode == config.eAndWGeoCode) {
+      comparatorCategory = get(englandAndWalesData).get(config.eAndWGeoCode).get(catCode);
```

Comment:
why does the england and wales store need to be indexed with a specific geocode? Won't there only ever be one geocode in there?